subsys/bluetooth/host/smp.c

```diff
@@ -3874,8 +3888,15 @@ static uint8_t smp_ident_addr_info(struct bt_smp *smp, struct net_buf *buf)
 			 * present before ie. due to re-pairing.
 			 */
 			if (!bt_addr_le_is_identity(&conn->le.dst)) {
+				struct addr_match addr_match = {
```

Comment:
Could you add a bsim test that exercises this part of the code? I'm fine with the current changes as long as they are tested.

Reply:
Added a BabbleSim test as you requested